frontend/server/src/main/java/org/pytorch/serve/util/PriorityLinkedBlockingDeque.java

```diff
@@ -19,70 +20,52 @@
     final ReentrantLock lock = new ReentrantLock();
     private final Condition notEmpty = lock.newCondition();
 
-    private int nPriorities;
-    private int sumPriorityWeights;
-    private int[] weightedPriorityMap;
-    private ConcurrentHashMap<Integer, LinkedBlockingDeque<T>> priorityDeques;
-
-    public PriorityLinkedBlockingDeque(int nPriorities, int queueSize) {
-
-        this.nPriorities = nPriorities;
-        this.priorityDeques = new ConcurrentHashMap<Integer, LinkedBlockingDeque<T>>();
-
-        // note: the weight calculation logic may be adapted to the user's needs
-        // sum of 0 + 1 + 2 + ... + nPriorities - 1 via triangular sum
-        this.sumPriorityWeights = ((this.nPriorities - 1) * this.nPriorities) / 2;
-        this.weightedPriorityMap = new int[sumPriorityWeights];
-
-        // initialize priority deques and weight map
-        int keyStart = 0;
-        for (int priority = 0; priority < this.nPriorities; priority++) {
-            this.priorityDeques.put(priority, new LinkedBlockingDeque<T>(queueSize));
-            if (priority > 0) {
-                // priority weights are inverse priority values for now
-                int priorityWeight = this.nPriorities - priority;
-                for (int key = keyStart; key < keyStart + priorityWeight; key++) {
-                    this.weightedPriorityMap[key] = priority;
-                }
-                keyStart += priorityWeight;
-            }
-        }
+    private int queueSize;
+    private float highPrioProb;
+    private ConcurrentHashMap<String, LinkedBlockingDeque<T>> priorityDeques;
+
+    public PriorityLinkedBlockingDeque(int queueSize, float highPrioProb) {
+
+        this.queueSize = queueSize;
```

Comment:
Is it necessary to store this? After all we never read that value. (and might be missleading, as the queue can store up to 3 times that value).

Reply:
True, we don't need to store it, I'll remove it. I agree that it might be misleading, but I'm not sure what would be the better way of doing it.

One option would be to rename the `queueSize` parameter to `sizePerQueue`, but that would break backwards compatibility, which I have been trying to avoid. Another option would be floordiv by 3, but that's also not so nice since the user will not end up with exactly the desired queue size. Do you have any ideas?

Reply:
I don't have any better idea either, so maybe lets keep it like that then.